### PR TITLE
Added editorconfig, updated typescript

### PR DESCRIPTION
**Pull request by amilajack**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
```
Comment (worr) [inline .editorconfig]: Take a look at the project; you'll notice that this isn't actually the style used throughout.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{json,js,ts}]
+indent_style = space
+indent_size = 2
```
Comment (worr) [inline .editorconfig]: See above comment about style

**Comment by amilajack** [comment]
My bad. I updated the editorconfig to match the existing project code style.

**Comment by amilajack** [comment]
@worr Also the build is failing because of:

```bash
...
tsc -m commonjs src/imdb.ts
mv src/imdb.js lib/imdb.js
mv: cannot move `src/imdb.js' to `lib/imdb.js': No such file or directory
make: *** [lib/imdb.js] Error 1
...
```

**Comment by worr** [comment]
Yes, this isn't anything that you directly changed, but it's failing because of new behavior in the `tsc` command. `tsc` used to drop the compiled file into the same directory as the source. That doesn't appear to be the case anymore. Can you update your patch to solve this problem?

Thanks!

**Comment by worr** [comment]
The new build machinery took care of the build failure itself.

Thanks for the contribution!

**Pull request merged**